src/JsonReader.cs

```diff
@@ -548,127 +570,162 @@ sealed class NonProperty : IJsonProperty<Unit, JsonReadResult<Unit>>
+                            else if (TrySetPropertyIndex(11, property11, ref reader, ref values)) { }
+                            else if (TrySetPropertyIndex(12, property12, ref reader, ref values)) { }
+                            else if (TrySetPropertyIndex(13, property13, ref reader, ref values)) { }
+                            else if (TrySetPropertyIndex(14, property14, ref reader, ref values)) { }
+                            else if (TrySetPropertyIndex(15, property15, ref reader, ref values)) { }
+                            else if (TrySetPropertyIndex(16, property16, ref reader, ref values)) { }
+
+                            if (!reader.Read())
+                                return reader.Suspend((sm, values));
+
+                            sm.OnPropertyNameRead();
 
-                    switch (property.Reader.TryRead(ref reader))
-                    {
-                        case { Incomplete: true }:
-                            throw PartialJsonNotSupportedException();
-                        case { Error: { } err }:
-                            error = err;
-                            return false;
-                        case { Value: var val }:
-                            value = (true, val);
-                            return true;
+                            break;
+
+                        case ObjectReadStateMachine.ReadResult.PropertyValue:
+                            static JsonReadResult<TResult>? ReadPropertyValue<TValue>(IJsonProperty<TValue, JsonReadResult<TValue>> property,
+                                                                                      ref Utf8JsonReader reader,
+                                                                                      ref (bool, TValue) value,
+                                                                                      ref ObjectReadStateMachine sm,
+                                                                                      ref ObjectValueState<T1, T2, T3, T4, T5, T6, T7, T8, T9, T10, T11, T12, T13, T14, T15, T16> values)
+                            {
+                                switch (property.Reader.TryRead(ref reader))
+                                {
+                                    case { Incomplete: true }:
+                                        return reader.Suspend((sm, values));
+                                    case { Error: { } err }:
+                                        return new JsonReadError(err);
+                                    case { Value: var val }:
+                                        value = (true, val);
+                                        return null;
+                                }
+                            }
+
+                            if (values.CurrentPropertyIndex is { } nextPropertyIndex)
+                            {
+                                var error = nextPropertyIndex switch
```

Comment:
Shouldn't `error` here be called `result`?

Reply:
Done in 5098ea3b82a6eec2c22ed0c7c37e3570fb2489df